src/assets/js/main.js

```diff
@@ -85,20 +85,55 @@ function scrollSidebarIntoView() {
   }
 }
 
-function adjustToc() {
-  // Adjustments to the jekyll-toc TOC.
+/**
+ * Adjusts the behavior of the table of contents (TOC) on the page.
+ * 
+ * This function enables a "scrollspy" feature on the TOC, where the active link in the TOC is updated
+ * based on the currently visible section in the page.
+ * 
+ * Enables a "back to top" button in the TOC header.
+ * returns {void}
+ */
+function adjustTOC() {
 
   const tocId = '#site-toc--side';
 
   const tocHeader = document.querySelector(tocId + ' header');
 
   if (tocHeader) {
-    tocHeader.addEventListener('click', function (_) {
-      $('html, body').animate({scrollTop: 0}, 'fast');
+    //Shouldn't it be a "to top" button? 
+    //Its doesn't make sense, it's not intutive.
+    tocHeader.addEventListener('click', (_) => {
+      _scrollToTop();
     });
   }
 
-  $('body').scrollspy({ offset: 100, target: tocId });
+  //Appear not to be used anywhere
+  //'class' active dont make any effect on toc
```

Comment:
On the live site, such as on https://docs.flutter.dev/ui/layout, it highlights the currently visible header on the page in grey instead of the default blue within the TOC. 

The scrollspy functionality to do this comes from bootstrap and I think for now it's fine to keep using it through jquery since our TOC is built with that in mind. Perhaps just add a comment explaining this.

We have to keep jquery until we migrate away from Bootstrap 4 anyway (https://github.com/flutter/website/issues/8030), so we can migrate this usage once we do that.